src/lib/io/sol-i2c-impl-linux.c

```diff
@@ -968,8 +968,22 @@ create_device_iter_cb(void *data, const char *dir_path, struct dirent *ent)
     char path[PATH_MAX];
     int r;
     struct stat st;
+#ifdef USE_I2C
+	unsigned int i2c_bus;
+	char *end_ptr;
+#endif
 
     if (strstartswith(ent->d_name, "i2c-")) {
+	#ifdef USE_I2C
+		errno = 0;
+        i2c_bus = strtoul(ent->d_name + strlen("i2c-"), &end_ptr, 0);
+        if (errno || *end_ptr != '\0') {
+			SOL_WRN("Could not get I2C bus number");
```

Comment:
If this happens, you should considerer an error and abort everything, don't you?